packages/build-tools/src/common/setup.ts

```diff
@@ -85,16 +87,33 @@ async function runExpoDoctor<TJob extends Job>(ctx: BuildContext<TJob>): Promise
   ctx.logger.info('Running "expo doctor"');
   let timeout: NodeJS.Timeout | undefined;
   let timedOut = false;
-  const argsPrefix = (await isAtLeastNpm7Async()) ? ['-y'] : [];
+  const isAtLeastNpm7 = await isAtLeastNpm7Async();
   try {
-    const promise = spawn('npx', [...argsPrefix, 'expo-doctor'], {
-      cwd: ctx.reactNativeProjectDirectory,
-      logger: ctx.logger,
-      env: ctx.env,
-    });
-    timeout = setTimeout(() => {
+    let promise: SpawnPromise<SpawnResult>;
+    if (!shouldUseGlobalExpoCli(ctx)) {
+      const argsPrefix = isAtLeastNpm7 ? ['-y'] : [];
+      promise = spawn('npx', [...argsPrefix, 'expo-doctor'], {
+        cwd: ctx.reactNativeProjectDirectory,
+        logger: ctx.logger,
+        env: ctx.env,
+      });
+    } else {
+      promise = ctx.runGlobalExpoCliCommand(
+        ['doctor'],
+        {
+          cwd: ctx.reactNativeProjectDirectory,
+          logger: ctx.logger,
+          env: ctx.env,
+        },
+        isAtLeastNpm7
+      );
+    }
+    timeout = setTimeout(async () => {
       timedOut = true;
-      promise.child.kill();
+      const ppid = nullthrows(promise.child.pid);
+      (await getAllChildrenRecursiveAsync(ppid)).forEach((pid) => {
```

Comment:
IMO the name is misleading. I wouldn't expect this function to return the parent process pid.
- If you want the function to return the parent pid as well, name it `getParentAndChildrenPidsAsync`.
- If you want to create the list with parent pid and the result of the function here, name it `getChildrenPidsAsync`.